src/operations/blas1/WGAtomicReduction.hpp

```diff
@@ -119,22 +126,38 @@ WGAtomicReduction<operator_t, lhs_t, rhs_t>::eval(sharedT scratch,
                                       cl::sycl::plus<value_t>());
   }
   if (ndItem.get_local_id()[0] == 0) {
-    atomic_res += val;
+    if constexpr (!usmManagedMem) {
+      auto atomic_res =
+          cl::sycl::atomic_ref<value_t, cl::sycl::memory_order::relaxed,
+                               cl::sycl::memory_scope::device,
+                               cl::sycl::access::address_space::global_space>(
+              lhs_.get_data()[0]);
+      atomic_res += val;
+    } else {
+      auto atomic_res =
+          cl::sycl::atomic_ref<value_t, cl::sycl::memory_order::relaxed,
+                               cl::sycl::memory_scope::device,
+                               cl::sycl::access::address_space::generic_space>(
+              lhs_.get_data()[0]);
+      atomic_res += val;
+    }
```

Comment:
Maybe, we can simplify things as follows.

```suggestion
    constexpr cl::sycl::access::address_space addr_sp =
        usmManagedMem ? cl::sycl::access::address_space::generic_space
                      : cl::sycl::access::address_space::global_space;
    auto atomic_res =
        cl::sycl::atomic_ref<value_t, cl::sycl::memory_order::relaxed,
                             cl::sycl::memory_scope::device, addr_sp>(
            lhs_.get_data()[0]);
    atomic_res += val;
```

Reply:
Thank you @pgorlani, written this way it looks cleaner.